hpke/util.go

```diff
@@ -85,7 +85,21 @@ func (suite Suite) getSuiteID() (id [10]byte) {
 	return
 }
 
-func (suite Suite) KEMScheme() kem.AuthScheme { return suite.kemID.Scheme() }
+func (suite Suite) KDFExtract(secret, salt []byte) (pseudorandomKey []byte) {
+	return hkdf.Extract(suite.kdfID.hash(), secret, salt)
+}
+
+func (suite Suite) KDFExpand(pseudorandomKey, info []byte, outputLen uint) []byte {
+	output := make([]byte, outputLen)
+	rd := hkdf.Expand(suite.kdfID.hash(), pseudorandomKey, info)
+	_, err := io.ReadFull(rd, output)
+	if err != nil {
+		return nil
```

Comment:
`nil` is ambiguous, since the output length would be 0 if `outputLen == 0`. If this error might be caused by user error, then I would suggest propagating it. If not, let's panic instead.

Reply:
Let's panic. This is an internal error we can't recover from.

Reply:
That said, we also don't validate any of the inputs here. The output length, in particular, [needs to be less than the `255*ExtractSize()`](https://tools.ietf.org/html/rfc5869). Should we add that?

Reply:
add validation of input/output